doc/user/content/sql/create-source/postgres.md

```diff
@@ -0,0 +1,139 @@
+aliases:
+  - /sql/create-source/postgresql
+---
+
+{{% create-source/intro %}}
+This document details how to connect Materialize to a Postgres database.
+{{% /create-source/intro %}}
+
+## Syntax
+
+{{< diagram "create-source-postgres.svg" >}}
+
+Field | Use
+------|-----
+**MATERIALIZED** | Materializes the source's data, which retains all data in memory and makes sources directly selectable. **Currently required for all Postgres sources.** For more information, see [Materialized source details](#materialized-source-details).
+_src_name_  | The name for the source.
+**IF NOT EXISTS**  | Do nothing (except issuing a notice) if a source with the same name already exists. _Default._
+**HOST** | Postgres host. See [Host options](#host-options) for details.
+**PUBLICATION** _publication_name_ | Postgres publication.
+
+### Host options
+
+Field | Type | Value
+------|------|------
+_host_  | Host for Postgres connection.
+_port_  | Port for Postgres connection.
+_user_  | User for Postgres connection.
+_password_  | Password for Postgres connection.
+_dbname_ | Postgres database name.
+_sslmode_ | Optionally specify TLS mode. Valid options are `disable`, `prefer`, `require`, `verify-ca`, and `verify-full`.
+_sslcert_ | SSL certificate.
+_sslkey_  |  SSL key.
+_sslrootcert_  | SSL root certificate.
+
+## Details
+
+### Materialized source details
+
+Materializing a source keeps data it receives in an in-memory
+[index](/overview/api-components/#indexes), the presence of which makes the
+source directly queryable. In contrast, non-materialized sources cannot process
+queries directly; to access the data the source receives, you need to create
+[materialized views](/sql/create-materialized-view) that `SELECT` from the
+source.
+
+For a mental model, materializing the source is approximately equivalent to
+creating a non-materialized source, and then creating a materialized view from
+all of the source's columns:
+
+```sql
+CREATE SOURCE src ...;
+CREATE MATERIALIZED VIEW src_view AS SELECT * FROM src;
+```
+
+The actual implementation of materialized sources differs, though, by letting
+you refer to the source's name directly in queries.
+
+For more details about the impact of materializing sources (and implicitly
+creating an index), see [`CREATE INDEX`: Details &mdash; Memory
+footprint](/sql/create-index/#memory-footprint).
+
+### PostgreSQL source details
+
+Materialize makes use of PostgreSQL's native replication capabilities to create a continuously updated replica of the desired Postgres tables.
+
+Before creating the source in Materialize, you must:
+
+- Set up your Postgres database to allow logical replication.
+- Ensure that the user for your Materialize connection has `REPLICATION` privileges.
+- Create a Postgres publication containing the data to be streamed to Materialize. Since Postgres sources are materialized (kept in memory) in their entirety, we strongly recommend that you limit publications only to the data you need to query.
```

Comment:
future work: it would be great to add a sentence or two explaining what a Postgres publication is, as i suspect most users won't have heard of this before this doc.